### ISSUE TITLE
Use token from LDAP instead of Token Role

### ISSUE THREAD

**Issue by monchaio** [Boostport/kubernetes-vault#155]
Hi

When I tried to use token that generate from other Authentication type like LDAP and set token to configmap.

I've always face error about token role from line of code:
https://github.com/Boostport/kubernetes-vault/blob/master/cmd/controller/client/vault.go#L261

So my questions is
1. Is it always need to use token role only for Kubernetes-Vault?
2. Could we add parameter to skip above role name checking?

Best regards,
Monchai

**Comment by F21** [comment]
Hey Monchai,

The reason we expected a role name is because the controller needs a periodic token, so that it can renew the token. Periodic tokens can be created using token roles and some a few auth methods: https://www.vaultproject.io/docs/concepts/tokens.html#periodic-tokens

Does the LDAP auth method support the creation of periodic token? If so, I think it should be possible to modify the code to not check the role name in certain cases.

**Comment by monchaio** [comment]
Hi @F21 

Yes I think LDAP authentication support the creation of periodic token.
Below is result when I execute command vault token lookup

```
Key                  Value
---                  -----
accessor             d5c7c2atAsy1iXdjXCTIAiqq
creation_time        1568951566
creation_ttl         24h
display_name         ldap-username
entity_id            3273bd33-4a70-0ee7-8134-1dde32f142dd
expire_time          2019-10-21T04:20:57.973333833+07:00
explicit_max_ttl     0s
id                   s.APXKt43sA5nvIzNnDa3fqvdO
issue_time           2019-09-20T10:52:46.899753954+07:00
last_renewal         2019-09-30T08:20:57.973334075+07:00
last_renewal_time    1569806457
meta                 map[username:username]
num_uses             0
orphan               true
path                 auth/ldap/login/username
policies             [default]
renewable            true
ttl                  499h59m25s
type                 service
```

Could I modify your code and create PR for this changes.

BR,
Monchai

**Comment by F21** [comment]
Hi @monchaio ,

Sure, please go ahead and create a PR. Maybe call the parameter `skip_token_role_name_validation`. It needs to be documented so that users can understand they should only use this if they really need to.

I was looking at the Vault API, hoping there would be a generalized method to check if a token is periodic, but I think we will need to implement this check for every auth backend. For example, for the token backend, it's part of a token role, for LDAP, it's part of the config, so it would be virtually impossible to implement, given that it's not possible to tell which auth backend created the token.

**Comment by monchaio** [comment]
Hi @F21 

Will create PR for this issue. Thanks so much for answer my question

Best regards
Monchai

**Comment by monchaio** [comment]
Hi @F21 

Sorry for late reply I just created PR https://github.com/Boostport/kubernetes-vault/pull/157
To add new parameter skipTokenRoleNameValidation.

Could you please kindly review?

Best regards,
Monchai

**Comment by F21** [comment]
Fixed in #157
